### PR TITLE
Projects Page Welcome Section

### LINKED ISSUE #2563: Implement a Welcome Section for the Projects Page
### **User Story**
As a user visiting the projects page, I want to see a collapsible "Welcome" section with tutorial videos so that I can quickly understand how to use the platform through Arkade videos. This section should allow users to effortlessly show or hide its content based on their preferences.

### **Details**
- **Header Title**: The section should have a warm and friendly title, like **_Good morning, Frank!_**
  
- **Tutorial Links**: 
   - Include 4 items (links to Arkade videos) styled as cards. We think the components from [this Tailwind UI blog page](https://tailwindui.com/components/marketing/sections/blog-sections#component-7c4bcd0e50e14e855d55d20b6cd445d1) would work well, and we’d like them to be rectangular in shape.
   - Each card title should clearly reflect the purpose of the corresponding Arkade tutorial.
  
- **Card Interaction**: 
   - Each card should be clickable, opening a modal with the Arkade video tutorial set to autoplay.
   - The modal must include a title and a close button.
   - Ensure that videos do not continue playing in the background once the modal is closed.

- **Collapsible Section**: 
   - The section should be collapsible, featuring an icon or arrow to indicate the ability to show/hide the content. By default, the section should be expanded.
   - When collapsed, only the section title and the expand/collapse icon should be visible. 
   - Implement functionality to save the user's preference for collapsing or expanding the section, ensuring it remains in the chosen state during future visits (integrate with issue #2564).

- **Tutorial Completion**: 
   - After a user finishes watching an Arkade tutorial, the associated card should display a visual cue (like a shade) indicating its completion. 
   - This preference should also be saved to remember the user's progress during their next visit.

### PR DESCRIPTION
**Pull request by elias-ba**
### Description

This PR introduces a collapsible "Welcome" section on the Projects page, designed to help users quickly understand the platform with Arkade tutorial videos. Users can easily show or hide this section based on their preferences. It also saves the user’s preferences for section visibility ensuring a seamless user experience across visits.

Closes #2563 & #2564

### Validation steps

1. **Visit the Projects Page:**  
   - Confirm that the "Welcome" section appears expanded by default with the friendly greeting header.

2. **Collapse/Expand Section:**  
   - Click the arrow/icon to collapse and expand the section.
   - Refresh the page and verify that the section maintains the last saved state (expanded/collapsed).

3. **Play Tutorial Videos:**  
   - Click a card to open the video modal. Ensure the video starts autoplaying.
   - Click a way to close the modal.

## AI Usage

Please disclose how you've used AI in this work (it's cool, we just want to know!):

- [ ] Code generation (copilot but not intellisense)
- [ ] Learning or fact checking
- [ ] Strategy / design
- [x] Optimisation / refactoring
- [x] Translation / spellchecking / doc gen
- [ ] Other
- [ ] I have not used AI

You can read more details in our [Responsible AI Policy](https://www.openfn.org/ai#pull-request-templates)

### Pre-submission checklist

- [x] I have performed a **self-review** of my code.
- [x] I have implemented and tested all related **authorization policies**. (e.g., `:owner`, `:admin`, `:editor`, `:viewer`)
- [x] I have updated the **changelog**.
- [x] I have ticked a box in "AI usage" in this PR

### CODE REVIEW

File: lib/lightning/accounts.ex [hunk 1]
```diff
@@ -995,4 +1010,58 @@ defmodule Lightning.Accounts do
...
+      where: u.id == ^user_id,
+      select: fragment("?->>?", u.preferences, ^key)
+    )
+    |> Repo.one()
+    |> case do
+      "true" -> true
+      "false" -> false
+      value -> value
+    end
+  end
```
[the remark this answers is not in the document]
Comment (elias-ba) [reply lib/lightning/accounts.ex:1041]: This was me, tho. But yeah, that would be a solid improvement.

**Comment by stuartc** [comment]
This looks good guys, thanks. I want to cherry pick out a commit and will merge this later today.

File: test/lightning_web/live/dashboard_live_test.exs [hunk 2]
```diff
@@ -262,6 +277,47 @@ defmodule LightningWeb.DashboardLiveTest do
...
+      assert view
+             |> has_element?("#arcade-banner-content[class~='max-h-[500px]']")
+
+      # Click to collapse the banner
+      view
+      |> element("button[phx-click='toggle-arcade-banner']")
+      |> render_click()
+
+      # Assert that the banner is now collapsed
+      refute view
```
Comment (elias-ba) [inline test/lightning_web/live/dashboard_live_test.exs:294]: @stuartc isn't this the reason why the tests were still passing ? Is it `~=` or `=~` ?

File: lib/lightning/accounts.ex (hunk 1 above)
[the remark this answers is not in the document]
Comment (midigofrank) [reply lib/lightning/accounts.ex:1041]: Sorry I'm late to the party. Something has just crossed my mind, do we really need to go to the db for this?

The `User` struct already has all the information we need, right?

[the remark this answers is not in the document]
Comment (elias-ba) [reply lib/lightning/accounts.ex:1041]: I think you are absolutely right. This is a nice catch. Given that we are not dealing with deeply nested JSON structures, then doing few `Map.get/3` is way better than dealing with database calls anytime the user opens that page. I wil push a change. Thanks

File: test/lightning_web/live/dashboard_live_test.exs (hunk 2 above)
Comment (stuartc) [reply test/lightning_web/live/dashboard_live_test.exs:294]: Ah no, it's correct, I didn't update the templates properly - just pushed. Thanks! 👍

**Pull request merged**